invokeai/frontend/web/src/features/parameters/types/parameterSchemas.ts

```diff
@@ -221,12 +229,18 @@ export type BaseModelParam = z.infer<typeof zBaseModel>;
 export const zMainModel = z.object({
   model_name: z.string().min(1),
   base_model: zBaseModel,
+  model_type: zModelType,
 });
 
 /**
  * Type alias for model parameter, inferred from its zod schema
  */
 export type MainModelParam = z.infer<typeof zMainModel>;
+
+/**
+ * Type alias for model parameter, inferred from its zod schema
+ */
+export type OnnxModelParam = z.infer<typeof zMainModel>;
```

Comment:
Schemas with single valid values for a property should just have the literal there. 

If the main model schema needs `model_type` now, it should have `model_type: z.literal("main")`. Same for onnx model schema. Otherwise the client will accept an onnx or sd main model in either situation.